File: src/main/kotlin/com/linqingying/cangjie/psi/CjTypeParameterListOwner.java
package com.linqingying.cangjie.psi;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;


public interface CjTypeParameterListOwner extends CjNamedDeclaration {
    @Nullable
    CjTypeParameterList getTypeParameterList();

    @Nullable
    CjTypeConstraintList getTypeConstraintList();

    @NotNull
    List<CjTypeConstraint> getTypeConstraints();

    @NotNull
    List<CjTypeParameter> getTypeParameters();
}

File: src/main/kotlin/com/linqingying/cangjie/psi/CjTypeParameterListOwner.kt
package com.linqingying.cangjie.psi

interface CjTypeParameterListOwner : CjNamedDeclaration {
    val typeParameterList: CjTypeParameterList?

    val typeConstraintList: CjTypeConstraintList?




    val typeConstraints: List<CjTypeConstraint >


    val typeParameters: List<CjTypeParameter >

}
interface CjTypeParameterListOwnerForExtend : CjTypeParameterListOwner {
    val extendTypeParameters: List<CjTypeParameter >
    val extendTypeConstraintList: CjTypeConstraintList?




    val extendTypeConstraints: List<CjTypeConstraint >

}
